#!/usr/bin/env python
PACKAGE = "tony_local_planner"

from dynamic_reconfigure.parameter_generator_catkin import *
from math import pi

gen = ParameterGenerator()

gen.add("max_vel_x", double_t, 0, "The maximum x velocity for the robot in m/s", 0.5, 0)
gen.add("max_vel_y", double_t, 0, "The maximum y velocity for the robot in m/s", 0.5, 0)
gen.add("max_vel_move_theta", double_t, 0, "The absolute value of the maximum rotational velocity when moving for the robot in rad/s",  0.5, 0)

gen.add("xy_goal_tolerance", double_t, 0, "Within what maximum distance we consider the robot to be in goal", 0.1, 0)
gen.add("yaw_goal_tolerance", double_t, 0, "Within what maximum angle difference we consider the robot to face goal direction", 0.05, 0)
gen.add("move_yaw_goal_tolerance", double_t, 0, "Within what maximum angle difference we consider the robot to face goal direction when moving", 0.0, 0)

gen.add("max_lookahead_distance", double_t, 0, "max lookahead distance in meter", 10, 0)
gen.add("max_available_curvature", double_t, 0, "max tolerable curvature", 10, 0)

gen.add("off_track_condition", double_t, 0, "off_track_condition before stop", 1.0, 0)
gen.add("off_track_error_condition", double_t, 0, "Local planner will fail if distance to the line exceeded this", 5.0, 0)

gen.add("transform_tolerance", double_t, 0, "transform_tolerance", 0.5, 0)
gen.add("steady_state_criterion", int_t, 0, "Steady state time represented using sampling time as time unit", 20, 1, 100)

gen.add("virtual_heading", double_t, 0, "virtual_heading (Deprecated, exists only for backward compatibility)", 0, -2*pi, 2*pi)

gen.generate(PACKAGE, "tony_local_planner", "TonyLocalPlanner")